Automattic/simplenote-ios
# Keyboard Support: MK2

### Details
In this PR we're adding the following features:
* shortcuts support in the note editor
* in-memory caching of scroll position for each note
* RTL support for keyboard navigation
* Tracks events for keyboard shortcuts

Closes #1092

### Test (iPhone / iPad with external keyboard)
1. Open a note

- [x] If markdown is enabled, preview can be toggled with `Cmd + Shift + P`
- [x] Toggle checklist with `Cmd + Shift + C`
- [x] If editor is not active, `left` arrow brings you back to the list
- [x] If editor is not active, `Cmd + Return` brings you back to the list

---

1. Open a note from the search

- [x] It is possible to navigate through matches with `Cmd + G` and `Cmd + Shift + G`

---

- [x] Notes remember its scroll position as you switch between different notes
- [x] When apps interface is RTL, left and right arrows work as expected
- [x] When using a shortcut there is a `Tracked: shortcut_used` printed in the console
- [x] On the note list: when search bar is active, pressing `Cmd + Shift + F` cancels search

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 1033eff0 with:

> Added support for keyboard navigation and shortcuts #1092
